### DISCUSSION TITLE
How to reuse custom react hooks in mobx Store

### DISCUSSION THREAD

**Discussion by lavcraft** [mobxjs/mobx#2892] [Q&A]
Hi, i'm using next-auth project for authentication. Also i love mobx :)

Sometimes this combination confusing me, because i really want use authenticated username in mobx store, but this information provided by next-auth react hook `useSession()`.

Typical page code:

```typescript
export const Page = () => {
   const [session, loading] = useSession();
   const store = useMyStore();
   
   return <div> 
      {session.user.email}
      {store.user.email}
   </div>
}
```

What is the best for way combine useSession and mobx store?

Advantages of migrating session to Mobx:

1. Use session anywhere in my app
2. Reactive handling UI changes when user authenticating (Change controls and other)
3. Avoiding information duplication
4. Binding some actions to authenticating data

Thanks all :)

**Comment by mweststrate** [comment]
use useEffect to write the current session state to the store somewhere at
the root of your app / page?

**Reply by lavcraft** [reply]
Yes, it workaround, but it's not best way because i should do it in each page. Can i concentrate this logic in `useStore()`? And how :)
